### PR TITLE
Adds log message when a secrets group for a git repository doesn't yi…

### PR DESCRIPTION
**Pull request by Kircheneer**
…eld a token.

# Closes: DNE
# What's Changed

When a git repository with an attached secrets group is synced today, but doesn't yield a token, there is no user feedback. This can make it frustrating to debug this case.

# TODO
- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [x] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/extras/datasources/git.py [hunk 1]
```diff
@@ -241,6 +241,13 @@ def get_repo_from_url_to_path_and_from_branch(repository_record):
         except ObjectDoesNotExist:
             # No defined secret, fall through to legacy behavior
             pass
+        if not token:
+            log_message = "Repository has a secrets group assigned but is missing a token secret of access type 'HTTP'. Falling through to legacy behaviour."
```
Comment (glennmatthews) [inline nautobot/extras/datasources/git.py:245]: Perhaps?

```suggestion
            log_message = (
                "Repository has a secrets group assigned but is missing a 'token' secret of access type 'HTTP'. "
                "Falling through to legacy behaviour."
            )
```

File: nautobot/extras/datasources/git.py [hunk 2]
```diff
@@ -310,7 +319,7 @@ def git_repository_dry_run(repository_record, job_result=None, logger=None):  #
         job_result (JobResult): Optional JobResult to store results into.
         logger (logging.Logger): Optional Logger to additionally log results to.
     """
-    from_url, to_path, from_branch = get_repo_from_url_to_path_and_from_branch(repository_record)
+    from_url, to_path, from_branch = get_repo_from_url_to_path_and_from_branch(repository_record, logger=logger)
```
Comment (glennmatthews) [inline nautobot/extras/datasources/git.py:322]: Why not pass `job_result` through in this case?

**Review by glennmatthews — COMMENTED** [review]
I like the fix here, though it'll need a bit of rework in `next` for 2.0 where the legacy behavior no longer exists. Thanks!

File: nautobot/extras/datasources/git.py [hunk 3]
```diff
@@ -241,6 +243,16 @@ def get_repo_from_url_to_path_and_from_branch(repository_record):
         except ObjectDoesNotExist:
             # No defined secret, fall through to legacy behavior
             pass
+        if not token:
+            log_message = (
+                "Repository has a secrets group assigned but is missing a token secret of access type 'HTTP'."
```
Comment (glennmatthews) [inline nautobot/extras/datasources/git.py:248]:
```suggestion
                "Repository has a secrets group assigned but is missing a 'token' secret of access type 'HTTP'."
```

**Comment by Kircheneer** [comment]
Note that I ignored the pylint error as is the established pattern in the file (see [here](https://github.com/nautobot/nautobot/pull/3897/files#diff-560821d3266ff4bf93a11d6a8ba0dd831bc4417a619607845749954159b07259L264))

**Pull request merged**